### ISSUE TITLE
Model with different result when running with Myriad [Bug]

### ISSUE THREAD

**Issue by PCPJ** [openvinotoolkit/openvino#9650]
##### System information (version)

- OpenVINO => 
- Operating System / Platform => Ubuntu 20.04.3 LTS, running inside a docker container (that is being hosted by a Ubuntu 20.04.3 LTS)
- Problem classification: Model Conversion
- Framework: PyTorch
- Model name: MobileNet_v2 (with some modifications)

##### Detailed description

I have two multi-label classification models, **A** and **B**, that I trained by myself.
Model **A** is performing much better than model **B**.
The problem I am facing is that both models are losing precision when I ran at the Myriad (I am using a [Luxonis OAK-1-PCBA](https://shop.luxonis.com/collections/all/products/bw1093)).
With losing precision I mean: running the same model at the same input image with different hardware (nvidia-gpu, intel-cpu, and Myriad) and receiving output scores that aren't the same.
This loss of precision is relevant at model **B**, and is huge at model **A**.
Here are the scores I received from model **A** and **B** at the same image with 4 different hardware:

Model **A** - Nvidia (using pytorch)
![Image_ModelA_NvidiaGPU](https://user-images.githubusercontent.com/8596365/149402125-6efc0f32-cb42-444a-8e01-7166919c4de0.jpg)

Model **A** - CPU (using the model converted to xml and the OpenVino Engine)
![Image_ModelA_OpenVinoCPU](https://user-images.githubusercontent.com/8596365/149392199-986c9364-845f-4d1c-aacd-f91c198310b4.jpg)

Model **A** - Myriad (using the model converted to xml and the OpenVIno Engine)
![Image_ModelA_openVinoMYRIAD](https://user-images.githubusercontent.com/8596365/149392426-d7c6f3aa-9386-46ca-946e-bdcdff97bfe6.jpg)

Model **A** - Myriad (using the model converted to blob and the Luxonis API)
![Image_ModelA_LuxonisAPI_MYRIAD](https://user-images.githubusercontent.com/8596365/149392487-c7f5fc7f-5002-4955-962e-614560966248.jpg)

Model **B** - Nvidia (using pytorch)
![Image_ModelB_NvidiaGPU](https://user-images.githubusercontent.com/8596365/149392994-e558e988-b4a0-4ea6-b106-be6ae8dfeeba.jpg)

Model **B** - CPU (using the model converted to xml and the OpenVino Engine)
![Image_ModelB_OpenVinoCPU](https://user-images.githubusercontent.com/8596365/149393053-6cccf0ac-3bae-4949-b092-31a5d43c1085.jpg)

Model **B** - Myriad (using the model converted to xml and the OpenVino Engine)
![Image_ModelB_OpenVinoMYRIAD](https://user-images.githubusercontent.com/8596365/149393106-285b8289-fef1-4a70-8908-0d347da732ad.jpg)

Model **B** - Myriad (using the model converted to blob and the Luxonis API)
![Image_ModelB_LuxonisAPI_MYRIAD](https://user-images.githubusercontent.com/8596365/149393135-a0b52e16-a10c-41b0-bc3f-cb457842279c.jpg)

##### Steps to reproduce
I am sharing the python class I am using to generate the results with the OpenVino engine.
Also sharing the models converted to IR (xml/bin) and to blob.
[Gogle drive link](https://drive.google.com/drive/folders/1tlFuwrKSnFIxeEs8ZK2w6I7vrTo7SVJM?usp=sharing)

The commands I am using to convert the models are:
```
python3 mo.py --input_model ModelA.onnx --input_shape [1,3,480,640] --data_type=FP16 --scale_values=[58.395,57.120,57.375] --mean_values=[123.675,116.28,103.53] --reverse_input_channels --move_to_preprocess --disable_fusing --disable_resnet_optimization
```
```
./myriad_compile -m ModelA.xml -ip U8 -VPU_NUMBER_OF_SHAVES 6 -VPU_NUMBER_OF_CMX_SLICES 6
```

The reason I am using the last parameters at the `mo.py` is to test if it would reduce the difference in the results. It didn't.
I converted the pytorch model to ONNX already using half precision, and the test running at the Nvidia-gpu used 16bits float numbers.

This is the input image I used for this issue:
![2021-11-5_09_43_18 846](https://user-images.githubusercontent.com/8596365/149400110-84458ee7-6e9c-41fb-b270-d2315dd6c44b.jpg)

If someone need mode pieces of code to reproduce more of this experiment I can share and explain them further.

Thank you so much in advance

**Comment by PCPJ** [comment]
PS: I didn't described with details the architecture from both models, but if someone thinks that it can be relevant I can explain it with no problems.

**Comment by avitial** [comment]
@PCPJ apologies for the delay providing a response, did a quick test using the public [mobilenet-v2-pytorch](https://github.com/openvinotoolkit/open_model_zoo/tree/master/models/public/mobilenet-v2-pytorch) with pre-trained model from OMZ and results for top 4 classes and probs across CPU, GPU and MYRIAD devices are comparable, refer to them below. 

I can see Model-B is way better than Model-A on MYRIAD device (when comparing results across devices) , when you state "MobileNet_v2 (with some modifications)", how different are your models compared to original mobilenet-v2 from PyTorch? And which version of OpenVINO were you using at the time of executing your test? I can see [Luxonis OAK-1-PCBA](https://shop.luxonis.com/collections/all/products/bw1093) is compatible with RPI, if still applicable please provide additional details of this setup (OS, platform, OpenVINO version, etc.).


<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/leavitia/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/leavitia/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">



</head>

<div style='border-width:100%'>

<body link="#0563C1" vlink="#954F72">



CPU FP16 | GPU FP16 | MYRIAD FP16 |   | CPU FP32 | GPU FP32 | MYRIAD FP32
-- | -- | -- | -- | -- | -- | --
classid probability | classid probability | classid probability |   | classid probability | classid probability | classid probability
450     11.0245209 | 450     11.1406250 | 450     10.8203125 |   | 450     11.0152588 | 450     11.0152540 | 450     10.8750000
573     9.3104267 | 573     9.4687500 | 573     8.8281250 |   | 573     9.3002129 | 573     9.3002033 | 573     8.7343750
627     8.6400537 | 627     8.6953125 | 627     8.4687500 |   | 627     8.6522799 | 627     8.6522837 | 627     8.4921875
468     8.5157518 | 468     8.6015625 | 458     8.0390625 |   | 468     8.4928532 | 468     8.4928493 | 458     8.0937500



</body>

</html>

**Comment by jgespino** [comment]
Closing, please re-open if additional assistance is needed.
